src/storage/oos_file.cpp

```diff
@@ -0,0 +1,433 @@
+
+    if (err != NO_ERROR)
+      {
+	return err;
+      }
+
+    PGSLOTID slotid = NULL_SLOTID;
+    int sp_status = spage_insert (thread_p, page_ptr.get(), &oos_rec, &slotid);
+    if (sp_status != SP_SUCCESS)
+      {
+	oos_log ("oos_insert_within_page: spage_insert failed with status %d\n", sp_status);
+	return ER_FAILED;
+      }
+    assert (slotid != NULL_SLOTID);
+
+    oos_recently_inserted_oos_vpid_map.emplace (&oos_vfid, vpid);
+    oos_log ("recently inserted oos vpid map size: %zu\n", oos_recently_inserted_oos_vpid_map.size());
+
+    oid.pageid = vpid.pageid;
+    oid.slotid = slotid;
+    oid.volid = vpid.volid;
+  }
+  assert (oos_rec.data == nullptr); // should be freed by auto_freed_recdes_ptr
+
+  return NO_ERROR;
+}
+
+static int oos_read_across_pages (THREAD_ENTRY *thread_p, const VFID &oos_vfid, const OID &oid,
+				  const OOS_RECORD_HEADER &first_chunk_header, RECDES &recdes)
+{
+  int err = NO_ERROR;
+  const auto header = first_chunk_header;
+  const int total_size = first_chunk_header.total_size;
+  assert (first_chunk_header.chunk_index == 0);
+  assert (total_size > oos_get_max_chunk_size_within_page ());
+
+  oos_log ("oos_read_across_pages: total_size=%d\n", total_size);
+
+  err = recdes_allocate_data_area (&recdes, total_size);
+
+  if (err != NO_ERROR)
+    {
+      return err;
+    }
+  recdes.type = REC_HOME;
+  recdes.length = total_size;
+
+  int idx = 0;
+  OID current_chunk_oid = oid;
+  char *buf = recdes.data;
+  int total_read_size = 0;
+  while (current_chunk_oid.pageid != NULL_PAGEID)
+    {
+      OOS_RECORD_HEADER header;
+      RECDES chunk_recdes;
+      {
+	int err = oos_read_within_page (thread_p, oos_vfid, current_chunk_oid, chunk_recdes, header);
+	if (err != NO_ERROR)
+	  {
+	    return err;
+	  }
+	auto_freed_recdes_ptr defer_free_chunk_recdes (&chunk_recdes, recdes_free_data_area);
+
+	oos_log ("read_large: header = {total_size=%d, chunk_index=%d, next_chunk_oid={pageid=%d, slotid=%d}}\n",
+		 header.total_size,
+		 header.chunk_index, header.next_chunk_oid.pageid, header.next_chunk_oid.slotid);
+
+	total_read_size += chunk_recdes.length;
+	std::memcpy (buf, chunk_recdes.data, chunk_recdes.length);
+	buf += chunk_recdes.length;
+	oos_log ("oos_read_across_pages: read chunk index=%d, chunk_size=%d\n", idx, chunk_recdes.length);
+	current_chunk_oid = header.next_chunk_oid;
+      }
+      assert (chunk_recdes.data == nullptr); // should be freed by auto_freed_recdes_ptr
+    }
+
+  assert (total_read_size == total_size);
+  assert (buf == total_size + recdes.data);
+
+  return NO_ERROR;
+}
+
+static int oos_read_within_page (THREAD_ENTRY *thread_p, const VFID &oos_vfid, const OID &oid, RECDES &recdes,
+				 OOS_RECORD_HEADER &header_out)
+{
+  const auto [pageid, slotid, volid] = oid;
+  auto vpid = VPID{pageid, volid};
+
+  PAGE_PTR raw_ptr = pgbuf_fix (thread_p, &vpid, OLD_PAGE_IF_IN_BUFFER, PGBUF_LATCH_READ, PGBUF_UNCONDITIONAL_LATCH);
```

Comment:
버퍼풀에 없어도 디스크에서 읽어와야 하기 때문에
```suggestion
  PAGE_PTR raw_ptr = pgbuf_fix (thread_p, &vpid, OLD_PAGE, PGBUF_LATCH_READ, PGBUF_UNCONDITIONAL_LATCH);
```

Reply:
반영했습니다. 감사합니다.